File: tui/styles/theme.go
package styles

import (
	"slices"

	"github.com/charmbracelet/lipgloss"
)

type Theme struct {
	Name     string
	Primary  string
	Accent   string
	Muted    string
	Success  string
	Error    string
	Warning  string
	Fg       string
	Border   string
	Selected string
}

var (
	availableThemes = []Theme{
		{
			Name:     "violet",
			Primary:  "#7C3AED",
			Accent:   "#A78BFA",
			Muted:    "#6B7280",
			Success:  "#10B981",
			Error:    "#EF4444",
			Warning:  "#F59E0B",
			Fg:       "#F9FAFB",
			Border:   "#374151",
			Selected: "#1F2937",
		},
		{
			Name:     "forest",
			Primary:  "#0F766E",
			Accent:   "#2DD4BF",
			Muted:    "#6B7280",
			Success:  "#22C55E",
			Error:    "#EF4444",
			Warning:  "#F59E0B",
			Fg:       "#ECFDF5",
			Border:   "#334155",
			Selected: "#0F172A",
		},
		{
			Name:     "sunset",
			Primary:  "#C2410C",
			Accent:   "#FB923C",
			Muted:    "#78716C",
			Success:  "#16A34A",
			Error:    "#DC2626",
			Warning:  "#F59E0B",
			Fg:       "#FFFBEB",
			Border:   "#44403C",
			Selected: "#292524",
		},
		{
			Name:     "ocean",
			Primary:  "#0369A1",
			Accent:   "#38BDF8",
			Muted:    "#64748B",
			Success:  "#10B981",
			Error:    "#EF4444",
			Warning:  "#F59E0B",
			Fg:       "#F8FAFC",
			Border:   "#334155",
			Selected: "#0F172A",
		},
		{
			Name:     "paper",
			Primary:  "#1D4ED8",
			Accent:   "#0F766E",
			Muted:    "#6B7280",
			Success:  "#15803D",
			Error:    "#B91C1C",
			Warning:  "#B45309",
			Fg:       "#111827",
			Border:   "#CBD5E1",
			Selected: "#E2E8F0",
		},
	}
	currentThemeName string

	// Colors
	ColorPrimary  lipgloss.Color
	ColorAccent   lipgloss.Color
	ColorMuted    lipgloss.Color
	ColorSuccess  lipgloss.Color
	ColorError    lipgloss.Color
	ColorWarning  lipgloss.Color
	ColorFg       lipgloss.Color
	ColorBorder   lipgloss.Color
	ColorSelected lipgloss.Color

	// Base styles
	Bold   lipgloss.Style
	Muted  lipgloss.Style
	Accent lipgloss.Style

	// App title bar
	TitleBar lipgloss.Style

	// Active / inactive tab indicators
	TabActive   lipgloss.Style
	TabInactive lipgloss.Style

	// Panel borders
	Panel       lipgloss.Style
	PanelActive lipgloss.Style

	// Section headers inside panels
	SectionHeader lipgloss.Style

	// Status / help bar at bottom
	HelpBar lipgloss.Style

	// Status messages
	StatusOK  lipgloss.Style
	StatusErr lipgloss.Style

	// Quote items in reference list
	QuoteItem   lipgloss.Style
	QuoteNumber lipgloss.Style

	// Modal overlay
	Modal lipgloss.Style

	// Form labels
	FormLabel lipgloss.Style

	// Button (focused)
	ButtonFocused lipgloss.Style

	// Button (unfocused)
	ButtonNormal lipgloss.Style
)

func init() {
	ApplyTheme("violet")
}

func ThemeNames() []string {
	names := make([]string, 0, len(availableThemes))
	for _, theme := range availableThemes {
		names = append(names, theme.Name)
	}
	return names
}

func CurrentThemeName() string {
	return currentThemeName
}

func ApplyTheme(name string) {
	theme := resolveTheme(name)
	currentThemeName = theme.Name

	ColorPrimary = lipgloss.Color(theme.Primary)
	ColorAccent = lipgloss.Color(theme.Accent)
	ColorMuted = lipgloss.Color(theme.Muted)
	ColorSuccess = lipgloss.Color(theme.Success)
	ColorError = lipgloss.Color(theme.Error)
	ColorWarning = lipgloss.Color(theme.Warning)
	ColorFg = lipgloss.Color(theme.Fg)
	ColorBorder = lipgloss.Color(theme.Border)
	ColorSelected = lipgloss.Color(theme.Selected)

	Bold = lipgloss.NewStyle().Bold(true)
	Muted = lipgloss.NewStyle().Foreground(ColorMuted)
	Accent = lipgloss.NewStyle().Foreground(ColorAccent)

	TitleBar = lipgloss.NewStyle().
		Foreground(ColorPrimary).
		Bold(true).
		Padding(0, 1)

	TabActive = lipgloss.NewStyle().
		Foreground(ColorAccent).
		Bold(true).
		Underline(true)

	TabInactive = lipgloss.NewStyle().
		Foreground(ColorMuted)

	Panel = lipgloss.NewStyle().
		Border(lipgloss.RoundedBorder()).
		BorderForeground(ColorBorder).
		Padding(0, 1)

	PanelActive = lipgloss.NewStyle().
		Border(lipgloss.RoundedBorder()).
		BorderForeground(ColorPrimary).
		Padding(0, 1)

	SectionHeader = lipgloss.NewStyle().
		Foreground(ColorAccent).
		Bold(true).
		MarginBottom(1)

	HelpBar = lipgloss.NewStyle().
		Foreground(ColorMuted).
		Padding(0, 1)

	StatusOK = lipgloss.NewStyle().Foreground(ColorSuccess)
	StatusErr = lipgloss.NewStyle().Foreground(ColorError)

	QuoteItem = lipgloss.NewStyle().
		Foreground(ColorFg).
		PaddingLeft(1)

	QuoteNumber = lipgloss.NewStyle().
		Foreground(ColorAccent).
		Bold(true)

	Modal = lipgloss.NewStyle().
		Border(lipgloss.DoubleBorder()).
		BorderForeground(ColorPrimary).
		Padding(1, 2)

	FormLabel = lipgloss.NewStyle().
		Foreground(ColorAccent).
		Width(20)

	ButtonFocused = lipgloss.NewStyle().
		Foreground(ColorFg).
		Border(lipgloss.NormalBorder()).
		BorderForeground(ColorPrimary).
		Background(ColorPrimary).
		Padding(0, 2).
		Bold(true)

	ButtonNormal = lipgloss.NewStyle().
		Foreground(ColorMuted).
		Border(lipgloss.NormalBorder()).
		BorderForeground(ColorBorder).
		Padding(0, 2)
}

func resolveTheme(name string) Theme {
	names := ThemeNames()
	if !slices.Contains(names, name) {
		return availableThemes[0]
	}
	for _, theme := range availableThemes {
		if theme.Name == name {
			return theme
		}
	}
	return availableThemes[0]
}

File: desktop/frontend/src/theme.ts
export type ThemeName = "violet" | "forest" | "sunset" | "ocean" | "paper";

type ThemeTokens = {
  bg: string;
  bgStrong: string;
  panel: string;
  panel2: string;
  border: string;
  borderStrong: string;
  primary: string;
  accent: string;
  muted: string;
  fg: string;
  ok: string;
  error: string;
  shadow: string;
  colorScheme: "dark" | "light";
};

const themes: Record<ThemeName, ThemeTokens> = {
  violet: {
    bg: "#0f172a",
    bgStrong: "#0b1120",
    panel: "rgba(17, 24, 39, 0.92)",
    panel2: "rgba(31, 41, 55, 0.82)",
    border: "#374151",
    borderStrong: "rgba(167, 139, 250, 0.42)",
    primary: "#7c3aed",
    accent: "#a78bfa",
    muted: "#94a3b8",
    fg: "#f9fafb",
    ok: "#10b981",
    error: "#ef4444",
    shadow: "0 24px 80px rgba(2, 6, 23, 0.38)",
    colorScheme: "dark",
  },
  forest: {
    bg: "#071a17",
    bgStrong: "#041311",
    panel: "rgba(9, 24, 21, 0.92)",
    panel2: "rgba(15, 41, 35, 0.82)",
    border: "#29443f",
    borderStrong: "rgba(45, 212, 191, 0.42)",
    primary: "#0f766e",
    accent: "#2dd4bf",
    muted: "#9ca3af",
    fg: "#ecfdf5",
    ok: "#22c55e",
    error: "#ef4444",
    shadow: "0 24px 80px rgba(1, 10, 9, 0.38)",
    colorScheme: "dark",
  },
  sunset: {
    bg: "#1c0f0a",
    bgStrong: "#130905",
    panel: "rgba(33, 17, 12, 0.94)",
    panel2: "rgba(52, 28, 18, 0.82)",
    border: "#5c4033",
    borderStrong: "rgba(251, 146, 60, 0.44)",
    primary: "#c2410c",
    accent: "#fb923c",
    muted: "#d6b8a6",
    fg: "#fffbeb",
    ok: "#16a34a",
    error: "#dc2626",
    shadow: "0 24px 80px rgba(20, 8, 2, 0.4)",
    colorScheme: "dark",
  },
  ocean: {
    bg: "#081824",
    bgStrong: "#06111a",
    panel: "rgba(11, 25, 38, 0.92)",
    panel2: "rgba(18, 40, 56, 0.82)",
    border: "#334155",
    borderStrong: "rgba(56, 189, 248, 0.42)",
    primary: "#0369a1",
    accent: "#38bdf8",
    muted: "#94a3b8",
    fg: "#f8fafc",
    ok: "#10b981",
    error: "#ef4444",
    shadow: "0 24px 80px rgba(3, 9, 16, 0.38)",
    colorScheme: "dark",
  },
  paper: {
    bg: "#f8fafc",
    bgStrong: "#e2e8f0",
    panel: "rgba(255, 255, 255, 0.96)",
    panel2: "rgba(248, 250, 252, 0.94)",
    border: "#cbd5e1",
    borderStrong: "rgba(29, 78, 216, 0.28)",
    primary: "#1d4ed8",
    accent: "#0f766e",
    muted: "#64748b",
    fg: "#111827",
    ok: "#15803d",
    error: "#b91c1c",
    shadow: "0 24px 80px rgba(148, 163, 184, 0.3)",
    colorScheme: "light",
  },
};

export function applyTheme(themeName: string): void {
  const theme = themes[(themeName as ThemeName) in themes ? (themeName as ThemeName) : "violet"];
  const root = document.documentElement;
  root.style.setProperty("--bg", theme.bg);
  root.style.setProperty("--bg-strong", theme.bgStrong);
  root.style.setProperty("--panel", theme.panel);
  root.style.setProperty("--panel-2", theme.panel2);
  root.style.setProperty("--border", theme.border);
  root.style.setProperty("--border-strong", theme.borderStrong);
  root.style.setProperty("--primary", theme.primary);
  root.style.setProperty("--accent", theme.accent);
  root.style.setProperty("--muted", theme.muted);
  root.style.setProperty("--fg", theme.fg);
  root.style.setProperty("--ok", theme.ok);
  root.style.setProperty("--error", theme.error);
  root.style.setProperty("--shadow", theme.shadow);
  root.style.setProperty("color-scheme", theme.colorScheme);
  document.body.dataset.theme = themeName;
}

export function themeNames(): ThemeName[] {
  return Object.keys(themes) as ThemeName[];
}
